subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/FilteredConfigurationIntegrationTest.groovy

```diff
@@ -68,14 +68,19 @@ project(':child2') {
 task verify {
     doLast {
         println "file-dependencies: " + configurations.compile.files { it instanceof FileCollectionDependency }.collect { it.name }
-        println "file-dependencies resolved-config: " + configurations.compile.resolvedConfiguration.getFiles { it instanceof FileCollectionDependency }.collect { it.name }
-        println "file-dependencies artifacts: " + configurations.compile.resolvedConfiguration.lenientConfiguration.getArtifacts { it instanceof FileCollectionDependency }.collect { it.file.name }
         println "external-dependencies: " + configurations.compile.files { it instanceof ExternalDependency }.collect { it.name }
-        println "external-dependencies resolved-config: " + configurations.compile.resolvedConfiguration.getFiles { it instanceof ExternalDependency }.collect { it.name }
-        println "external-dependencies artifacts: " + configurations.compile.resolvedConfiguration.lenientConfiguration.getArtifacts { it instanceof ExternalDependency }.collect { it.file.name }
         println "child1-dependencies: " + configurations.compile.files { it instanceof ProjectDependency && it.dependencyProject.name == 'child1' }.collect { it.name }
+            }
+}
+
+task verifyViaResolvedConfiguration {
```

Comment:
Instead of duplicating this as two tasks, would it make sense to instead verify that the result of the resolvedConfiguration is the same as the non-resolved case?

That would allow us to keep the assertions below the same (we're only verifying the "good" APIs output) and keep the resolvedConfiguration and good API in sync.

Reply:
After examining this a little closer and looking at `ArtifactView`, I don't think we need to test AV here.  This is just using file names for each comparison, it's really comparing resolved dependencies so I don't think the AV is appropriate.